File: src/typescript/transaction-builder/operation-factories/comment-builder/creating-reply.ts
import { createHiveChain, ReplyOperation } from '@hiveio/wax';

// Initialize hive chain interface
const chain = await createHiveChain();

// Initialize a transaction object
const tx = await chain.createTransaction();

// Use multiple explicit values
tx.pushOperation(new ReplyOperation({
  // Here you can pass the arguments to given operation class constructor
  // Here `parentAuthor` and `parentPermlink` arguments can't be skipped nor be empty.
  // Also other required operation basic attributes (like `author`) must be explicitly specified.
  parentAuthor: 'parent-author',
  parentPermlink: 'parent-permlink',
  author: 'reply-author',
  permlink: 'reply-permlink',
  body: 'the-reply-body',
  description: 'This is the description of the post inside ReplyOperation',
  tags: ['hive']
}));

// Get a transaction object holding all operations and transaction TAPOS & expiration data, but transaction is **not signed yet**
console.log(tx.transaction);

File: src/rust/transaction/operations/comment/creating_reply.rs
use wax::prelude::*;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    // Initialize hive chain interface
    let chain = create_hive_chain(None)?;

    // Initialize a transaction object
    let mut tx = chain.create_transaction(None).await?;

    // Use multiple explicit values
    tx.push_complex_operation(
        &chain,
        ReplyOperation {
            // Here you can pass the arguments to given operation struct fields
            // Here `parent_author` and `parent_permlink` arguments can't be skipped nor be empty.
            // Also other required operation basic attributes (like `author`) must be explicitly specified.
            parent_author: "parent-author".into(),
            parent_permlink: "parent-permlink".into(),
            author: "reply-author".into(),
            permlink: Some("reply-permlink".into()),
            body: "the-reply-body".into(),
            description: Some(
                "This is the description of the post inside ReplyOperation"
                    .into(),
            ),
            tags: vec!["hive".into()],
            ..Default::default()
        },
    )?;

    /*
    Get a transaction object holding all operations and transaction
    TAPOS & expiration data, but transaction is **not signed yet**
    */
    println!("{:#?}", tx.transaction());

    Ok(())
}
